File: android/src/main/kotlin/com/example/pl_image_downloader/models/DownloadResult.kt
package com.example.pl_image_downloader.models



/**
 *  DownloadResult
 * Data class representing the result of a download operation.
 */
data class DownloadResult(
    /** The path where the file is saved. */
    val path: String,
    /** The directory where the file is saved. */
    val dictionary: String,
    /** The name of the downloaded file. */
    val fileName: String,
    /** Indicates whether the download was successful. */
    val isSuccess: Boolean,
    /** An optional error message if the download failed. */
    val errorMessage: String? = null
) {
    /** * Converts the DownloadResult instance to a Map representation. */
    fun toMap(): Map<String, Any> {
        val result = mapOf(
            "path" to path,
            "fileName" to fileName,
            "directoryResult" to dictionary,
            "isSuccess" to isSuccess,
        )
        errorMessage ?: return result
        return result + mapOf("errorMessage" to errorMessage)
    }
}

File: ios/Classes/Models/DownloadResult.swift
//
//  DownloadResult.swift
//  Developer
//
//  Created by Nguyễn Minh Hưng on 5/5/26.
//

public struct DownloadResult {
    var path: String = ""
    var dictionary: String = ""
    var fileName: String = ""
    var isSuccess: Bool = false
    var error: String? = null
}


public extension DownloadResult {
    func toMap() -> [String: Any] {
        var map = [String: Any](
            "path", path,
            "directoryResult", dictionary,
            "fileName", fileName,
            "isSuccess", isSuccess,
        )
        if let error = error {
            map["errorMessage"] = error
        }
        return map
    }
}
